python/crusher.py

```diff
@@ -0,0 +1,163 @@
+"""
+Crusher code by yiming liu
+
+This work is according to Mariia Fomicheva's theory in paper 'Micropolar medium in a funnel-shaped crusher' published in 2021
+
+"""
+
+from dolfin import *
+import mshr
+from ufl import indices
+import matplotlib.pyplot as plt
+%matplotlib inline
+import numpy
+import time
+'''
+1 1 "walls"
+1 2 "inlet"
+1 3 "outlet"
+2 1 "matter"
+
+'''
+#solver parameters settings according to Emek's work in 2019 'An Accurate Finite Element Method for the Numerical Solution 
+# of Isothermal and Incompressible Flow of Viscous Fluid'
+parameters["form_compiler"]["representation"] = "uflacs"
+parameters["form_compiler"]["cpp_optimize"] = True
+parameters["form_compiler"]["optimize"] = True
+parameters["allow_extrapolation"] = True
+parameters["mesh_partitioner"] = "SCOTCH"#"ParMETIS"
+parameters["ghost_mode"] = 'shared_vertex'
+
+krylov_params = parameters["krylov_solver"]
+krylov_params["relative_tolerance"] = 1E-3
+krylov_params["absolute_tolerance"] = 1E-6
+krylov_params["monitor_convergence"] = False
+krylov_params["maximum_iterations"] = 15000
+
+processID =  MPI.comm_world.Get_rank()
+set_log_level(50) #50, 40, 30, 20, 16, 13, 10
+
+foldername = "crusher_2w/"
+t_end = 1.05#8
+dt = 1./1000. #1600
+t=0.
+n=0
+
+fname = "geo/crusher_2w"
+mesh = Mesh(fname + '.xml')
+cells = MeshFunction('size_t', mesh , fname + '_physical_region.xml')
+facets = MeshFunction('size_t', mesh , fname + '_facet_region.xml')
+#nn = FacetNormal(mesh)
+#on the server benz, the xml cannot be read, How to deal with this problem?
+
+i, j, k, l = indices(4)
+delta = Identity(2)
+
+scalar = FiniteElement('P', triangle, 1)
+vector = VectorElement('P', triangle, 1)
```

Comment:
Here you need to use second order polynomials as well.

Reply:
After watching that video, I think I kind of understand why we use second order polynomials here. Thank you so much!